### PR TITLE
[RESTEASY-2574] create swagger integration example

### PR DESCRIPTION
**Pull request by Moicen**

### CODE REVIEW

**Comment by liweinan** [comment]
* [RESTEASY-2574 create swagger integration example](https://issues.redhat.com/browse/RESTEASY-2574)

**Comment by liweinan** [comment]
@Moicen please cleanup the README and translate chinese text into English,thanks!

**Comment by Moicen** [comment]
> @Moicen please cleanup the README and translate chinese text into English,thanks!

OK, done that.

**Comment by liweinan** [comment]
@Moicen please fix the depdency errors:

1. exclude the conflicted artifacts like this: https://github.com/resteasy/resteasy-spring-boot/blob/master/resteasy-spring-boot-starter/pom.xml#L138
2. add the dependency explicitly.

**Comment by Moicen** [comment]
> @Moicen please fix the depdency errors:
> 
> 1. exclude the conflicted artifacts like this: https://github.com/resteasy/resteasy-spring-boot/blob/master/resteasy-spring-boot-starter/pom.xml#L138
> 2. add the dependency explicitly.

Got it. I've made some changes, please review again, thanks.

**Comment by Moicen** [comment]
Hi @liweinan , I use `com.acme` as package name because `org.jboss.resteasy` is excluded by swagger plugin by default:[https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-integration/src/main/java/io/swagger/v3/oas/integration/IgnoredPackages.java](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-integration/src/main/java/io/swagger/v3/oas/integration/IgnoredPackages.java)

File: jaxrs-swagger-example/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+With this `jaxrs-api.json` file we can start a web service, to be convenient, we can use the official docker image of swagger:
```
Comment (liweinan) [inline jaxrs-swagger-example/README.md:67]: s/:/./

**Comment by liweinan** [comment]
@Moicen the pr looks good to me now. Thanks for your contribution.

In the mean time, we require that all contributions are made under the terms of the ASL2.0 License: http://www.apache.org/licenses/LICENSE-2.0 Do you agree to these terms?
Thanks!

File: pom.xml [hunk 2]
```diff
@@ -109,6 +110,12 @@
                 <artifactId>resteasy-spring-web</artifactId>
                 <version>${resteasy.ver}</version>
             </dependency>
+            <dependency>
```
Comment (liweinan) [inline pom.xml:113]: @Moicen please move this dependency into your own example module.

Comment (Moicen) [reply pom.xml:113]: ok

**Comment by Moicen** [comment]
> @Moicen the pr looks good to me now. Thanks for your contribution.
> 
> In the mean time, we require that all contributions are made under the terms of the ASL2.0 License: http://www.apache.org/licenses/LICENSE-2.0 Do you agree to these terms?
> Thanks!

yes, I agree.

**Comment by liweinan** [comment]
@Moicen  thanks.

@asoldano The PR looks good to me now. Could you please help to review this example?

**Comment by liweinan** [comment]
@asoldano with my understanding this example is independent from Spring framework.

@Moicen could you please help to confirm this? 

If that's the situation, please make the following changes:

1. Change the type of this project from JAR to WAR 
2. Use Jetty plugin instead of Spring-Boot Server.

**Comment by Moicen** [comment]
@liweinan confirmed. I'll remove all spring-* dependencies in this module and use jetty instead.

**Comment by liweinan** [comment]
@Moicen thanks! Let me verify it soon.

**Comment by liweinan** [comment]
@Moicen Thanks for the modifications. I can verify this example works locally:

Swagger UI service:

D89500A7-9E40-45BE-855C-8FA057243010

Jetty:

image

Access the service:

image

**Comment by Moicen** [comment]
@liweinan glad to hear that, thanks. I've updated the new README and fixed some informal descriptions, please check again.

**Comment by liweinan** [comment]
@Moicen  Thanks, the example looks good to me now.

@asoldano Could you please help to review it again? Thanks!

**Comment by liweinan** [comment]
@asoldano thanks for reviewing!

@Moicen thanks for contributing!

**Pull request merged**